src/resonanceReconstruction/rmatrix/legacy/resolved/SpinGroup/SingleLevelBreitWigner/src/grid.hpp

```diff
@@ -0,0 +1,28 @@
+/**
+*  @brief Return the minimal energy grid derived from the resonance parameters
+*         for the spin group
+ *
+ *  The minimal energy grid consists of all resonance energy values (only the
+ *  ones that are positive) along with the resonance energies +/- half the
+ *  total width.
+ */
+std::vector< Energy > grid() const {
+
+  std::vector< Energy > grid;
+  grid.reserve( 3 * this->resonanceTable().numberResonances() );
```

Comment:
I'm guessing that `energy > 0. * electronVolt` most of the time. :-) I suppose you could always precompute the exact number of push_back()s you're going to get, if absolutely minimizing space is a priority.

Reply:
There's often only a few negative resonances - these are used by evaluators to adjust the value of the cross section. More often than not, 3 times the number of resonances is a good enough measure of the size of this vector.